### DISCUSSION TITLE
Use field name for relations table suffix when M2M field is linked to parent collection

### DISCUSSION THREAD

**Discussion by smokeyfro** [directus/directus#17215] [Feature Requests]
At the moment the M2M field relation table is [parent]_[related-collection], which makes total sense. However when relating a field to its parent collection the table ends up being [parent]_[parent] and if an additional related field is added it becomes [parent]_[parent]1.

For example, I have a plants collection, that includes general information about a plant, as well as which plants are companions and which should be avoided. In order to handle the relations, I'll be using the M2M tags extension, which works the same as the core M2M field, but inline. 

The plant collection has two M2M fields, one of friends and one for foes. The resulting join tables are plant_plant and plant_plant1. I realise this is really nitpicking here, but it would be really awesome if Directus would pickup that we're linking internally and in that case use the M2M field name as the suffix to the join table. 

For example plant_plant would become plant_friends and plant_plant1 would become plant_foes. 

I realize this is probably trivial, as these are hidden from the ui, but the OCD in me looks at the current table names and thinks, that really isn't clear which is for what.

**Comment by rijkvanzanten** [comment]
So just to confirm, this would only apply when you're making a recursive relationship right?

**Reply by smokeyfro** [reply]
Yeah :)

**Comment by smokeyfro** [comment]
![Screenshot from 2023-01-20 11-17-06](https://user-images.githubusercontent.com/109826/213659509-aa28cf1a-e1d2-4f40-9687-4a087479365d.png)

**Reply by smokeyfro** [reply]
I just changed the field types to M2A and noticed it uses the field name for the join table, so plants_friends and plants_foes, which is what i was hoping for with M2M. 

One small suggestion for the M2A field, if only one collection is assigned, skip the collection selection step (on the create / select buttons) and make it configurable to remove the collection prefix from the selected list.

**Reply by rijkvanzanten** [reply]
> One small suggestion for the M2A field, if only one collection is assigned, skip the collection selection step (on the create / select buttons) and make it configurable to remove the collection prefix from the selected list.

Out of personal curiosity, why would you use a M2A over a M2M in that case? The reason the M2A exists is to use multiple related collections 🤔

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
